app/screens/image_preview/image_preview.js

```diff
@@ -370,10 +378,24 @@ export default class ImagePreview extends PureComponent {
                 return this.renderVideoPreview(file);
             }
 
+            const isGifImage = isGif(file.data);
+            if (file.data.has_preview_image) {
+                ImageCacheManager.cache(file.data.name, Client4.getFilePreviewUrl(file.data.id), this.updateFileImage);
```

Comment:
This should happen somewhere like `componentDidUpdate`. `render` shouldn't have side effects like this

Reply:
not really cause this only happens when the Gallery fails to render the image

Reply:
I don't follow why it can't be done in `componentDidUpdate` though. We really shouldn't be making requests in `render`

Reply:
Oh, I didn't realize that this is for rendering errors in the Gallery component. I really don't like this, but I can't really see any way to make this work without making changes to that component